src/kernel/mconfig.ml

```diff
@@ -1027,7 +1037,11 @@ let unitname t =
   | Some name -> Misc.unitname name
   | None ->
     let basename = Misc.unitname t.query.filename in
+    (* CR: get rid of wrapping_prefix. it is only here for legacy reasons at the moment *)
     begin match t.merlin.wrapping_prefix with
     | Some prefix -> prefix ^ basename
-    | None -> basename
+    | None ->
+      String.Map.find_opt basename t.merlin.unit_name_for
+      |> Option.map ~f:Misc.unitname
```

Comment:
Why is `Misc.unitname` needed here when it wasn't for `wrapping_prefix`?

Reply:
Ultimately this doesn't matter much - `Misc.unitname` really just capitalized the first letter and drops a file extension if present. Really, I think maybe the `Misc.unitname` should've been included for the wrapping-prefix version.

Reply:
I added the `Misc.unitname` for wrapping prefix